packages/block-library/src/post-title/index.php

```diff
@@ -33,10 +33,19 @@ function render_block_core_post_title( $attributes, $content, $block ) {
 		$tag_name = 0 === $attributes['level'] ? 'p' : 'h' . $attributes['level'];
 	}
 
+	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $align_class_name ) );
+
 	if ( isset( $attributes['isLink'] ) && $attributes['isLink'] ) {
-		$title = sprintf( '<a href="%1$s" target="%2$s" rel="%3$s">%4$s</a>', get_the_permalink( $post_ID ), esc_attr( $attributes['linkTarget'] ), esc_attr( $attributes['rel'] ), $title );
+		return sprintf(
+			'<%1$s><a href="%2$s" target="%3$s" rel="%4$s" %5$s>%6$s</a></%1$s>',
```

Comment:
Where did `$tag_name` come from? Why do we now have a wrapping element?

Reply:
Previously this was output to a `$title` variable. `$title` was then wrapped in a wrapping element after the if statement.

This change simply returns the whole markup rather than creating a variable, which I thought was easier to read than having multiple if statements.

Reply:
I appreciate this might seem like a nit pick, but in order to keep this PR _focused_ I really think we should go back to the previous setup whereby we had a single `return` statement.

Why?

Having two `return` statements means the HTML output is effectively generated in x2 places thus increasingly the complexity of reasoning around the code. It also doubles the surface area for the introduction of XSS bugs (as evidenced by the original omission of the `esc_*` functions in this PR).